packages/core/src/modules/proofs/formats/presentation-exchange/PresentationExchangeFormatService.ts

```diff
@@ -0,0 +1,566 @@
+      uriList = [...uriList, ...inputDescriptor.schema.map((s) => s.uri)]
+    }
+
+    const query = []
+    for (const inputDescriptor of presentationDefinition.input_descriptors) {
+      for (const schema of inputDescriptor.schema) {
+        const innerQuery: Query<W3cCredentialRecord> = {}
+        innerQuery.$or = [{ expandedType: [schema.uri] }, { contexts: [schema.uri] }]
+        query.push(innerQuery)
+      }
+    }
+
+    // query the wallet ourselves first to avoid the need to query the pex library for all
+    // credentials for every proof request
+
+    const credentials = await this.w3cCredentialService.findCredentialRecordsByQuery(agentContext, {
+      $or: [...query],
+    })
+
+    const pexCredentials = credentials.map((c) => JsonTransformer.toJSON(c) as IVerifiableCredential)
+
+    const pex: PEXv1 = new PEXv1()
+    const selectResults = pex.selectFrom(presentationDefinition, pexCredentials)
+
+    if (selectResults.verifiableCredential?.length === 0) {
+      throw new AriesFrameworkError('No matching credentials found.')
+    }
+
+    return {
+      proofFormats: {
+        presentationExchange: {
+          formats: selectResults,
+        },
+      },
+    }
+  }
+
+  public async autoSelectCredentialsForProofRequest(
+    options: FormatRetrievedCredentialOptions<[PresentationExchangeProofFormat]>
+  ): Promise<FormatRequestedCredentialReturn<[PresentationExchangeProofFormat]>> {
+    const presentationExchange = options.proofFormats.presentationExchange
+    if (!presentationExchange) {
+      throw new AriesFrameworkError('No presentation options provided')
+    }
+
+    if (
+      !presentationExchange.formats.verifiableCredential ||
+      presentationExchange.formats.verifiableCredential.length === 0
+    ) {
+      throw new AriesFrameworkError('No credentials provided')
+    }
+
+    // check if this is INFO / (maybe WARNING. check when this is warning?)
+    presentationExchange.formats.areRequiredCredentialsPresent
+
+    // TODO SubmissionRequests...
+
+    // 100 credentials in the credential list
+    // 4 groups in total that satisfy the proof request
+    //  for each group I need to know which credentials it needs
+
+    // How to auto select the credentials:
+
+    //  1. loop over all matches and find the first match for each submission requirement
+    //  2. then for each match we extract the associated credentials from the `presentationExchange.verifiableCredential` array
+    // We probably also need to return the selected matches we used so we can use those to create the presentation submission
+
+    // Check how to correlate it. I think we may need to do something with the count here?
+
+    // note matches is array of SubmissionRequirementMatch
+    // presentationExchange.formats.matches[0].count
+    return {
+      proofFormats: {
+        presentationExchange: {
+          formats: presentationExchange.formats.verifiableCredential[0],
+        },
+      },
```

Comment:
This seems incomplete. Some notes:

1. areRequiredCredentialsPresent as input to this method doesn't seem right. We should do the validation in this method, or not do the validation.
2. This only returns a single credential. A presentation exchange request could request multiple credentials at the same time. Thsi doesn't seem sufficient.
3. Maybe we can come up with a better name than `formats` to nest it under? This is already under `proofFormats` , what does `formats` mean in this context?

Reply:
I wanted to leave the multiple credentials until post merge as this PR is already too big. I am not even sure how we do that so it needs a lot more work. 

Can we do this in a separate PR?